chore: test(acc test): Support multi-resource acceptance check

Lance52259 opened the pull request:
<!-- Thanks for sending a pull request! -->

**What this PR does / why we need it**:
The resource created by `count` will carry an index in the state, which cannot be verified by existing functions. e.g.:
`huaweicloud_evs_volume.test.0`, `huaweicloud_evs_volume.test.1`, ...

Through the expansion of existing functions, adding indexes to support the verification of multiple resources.

**Which issue this PR fixes**:
```
NONE
```

**Special notes for your reviewer**:

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->

```release-note
1. add multi-check methods to acceptance.
2. update acc tests of the evs volume.
```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/evs' TESTARGS='-run=TestAccEvsVolume_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/evs -v -run=TestAccEvsVolume_basic -timeout 360m -parallel 4
=== RUN   TestAccEvsVolume_basic
=== PAUSE TestAccEvsVolume_basic
=== CONT  TestAccEvsVolume_basic
--- PASS: TestAccEvsVolume_basic (148.86s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/evs       148.957s
```

```
make testacc TEST='./huaweicloud/services/acceptance/evs' TESTARGS='-run=TestAccEvsVolume_withEpsId'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/evs -v -run=TestAccEvsVolume_withEpsId -timeout 360m -parallel 4
=== RUN   TestAccEvsVolume_withEpsId
=== PAUSE TestAccEvsVolume_withEpsId
=== CONT  TestAccEvsVolume_withEpsId
--- PASS: TestAccEvsVolume_withEpsId (82.19s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/evs       82.304s
```

Lance52259 committed:
chore(acc/check): support multi-resource detection of exist checking

Jason-Zhang9309 commented on huaweicloud/services/acceptance/evs/resource_huaweicloud_evs_volume_test.go:
```diff
@@ -113,56 +133,70 @@ func TestAccEvsVolume_withEpsId(t *testing.T) {
   tags = {
     foo = "bar"
-    key = "value"
+	key = "value"
```
please align the equals signs

Lance52259 replied:
done

ShiChangkuo commented on huaweicloud/services/acceptance/acceptance.go:
```diff
@@ -196,14 +196,56 @@ func TestCheckResourceAttrWithVariable(resourceName, key, varStr string) resourc
+			continue
+		}
+
+		conf := TestAccProvider.Meta().(*config.Config)
```
suggest to put this line outside for loop.

Lance52259 replied:
done

ShiChangkuo commented on huaweicloud/services/acceptance/acceptance.go:
```diff
@@ -212,67 +254,91 @@ func (rc *resourceCheck) CheckResourceDestroy() resource.TestCheckFunc {
+  Parameter:
+    count: the expected number of resources that has been destroyed.
+*/
+func (rc *resourceCheck) CheckMultiResourceDestroy(count int) resource.TestCheckFunc {
```
s/CheckMultiResourceDestroy/CheckMultiResource**s**Destroy

Lance52259 replied:
done

ShiChangkuo commented on huaweicloud/services/acceptance/acceptance.go:
```diff
@@ -212,67 +254,91 @@ func (rc *resourceCheck) CheckResourceDestroy() resource.TestCheckFunc {
+/*
+CheckMultiResourceDestroy checks whether multiple resources created by count in HuaweiCloud are destroyed.
+  Parameter:
+    count: the expected number of resources that has been destroyed.
```
suggest to use ***expCount* instead

Lance52259 replied:
that's right

ShiChangkuo commented on huaweicloud/services/acceptance/acceptance.go:
```diff
@@ -212,67 +254,91 @@ func (rc *resourceCheck) CheckResourceDestroy() resource.TestCheckFunc {
+  Parameters:
+    count: the expected number of resources that will be created.
+*/
+func (rc *resourceCheck) CheckMultiResourceExists(count int) resource.TestCheckFunc {
```
s/CheckMultiResourceExists/CheckMultiResource**s**Exists

Lance52259 replied:
done

chengxiangdong commented on huaweicloud/services/acceptance/acceptance.go:
```diff
@@ -196,14 +196,56 @@ func TestCheckResourceAttrWithVariable(resourceName, key, varStr string) resourc
+	strs := strings.Split(rc.resourceName, ".")
+	var resourceType string
+	for _, str := range strs {
+		if strings.Contains(str, "huaweicloud_") {
```
a legacy issue, pls add a check below and return an error if `resourceType` is empty

Lance52259 replied:
OFC, the return has been added.

Lance52259 committed:
chore(acc test): optimized for acc test

- correct the description of the word destroy
- add a error return for invalid resource name
- adjust the code logic of loop

Lance52259 committed:
test(evs): support multi-resources checks in acc test

chengxiangdong commented on huaweicloud/services/acceptance/acceptance.go:
```diff
@@ -196,83 +196,137 @@ func TestCheckResourceAttrWithVariable(resourceName, key, varStr string) resourc
+
+	conf := TestAccProvider.Meta().(*config.Config)
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != resourceType {
```
`CheckResourceDestroy` does not need to be changed, it checks all resources that matched the `resourceType`

Lance52259 replied:
yeah, this function has been restored.

Lance52259 commented:
```
make testacc TEST='./huaweicloud/services/acceptance/evs' TESTARGS='-run=TestAccEvsVolume_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/evs -v -run=TestAccEvsVolume_basic -timeout 360m -parallel 4
=== RUN   TestAccEvsVolume_basic
=== PAUSE TestAccEvsVolume_basic
=== CONT  TestAccEvsVolume_basic
--- PASS: TestAccEvsVolume_basic (135.39s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/evs       135.485s
```
